columnflow/columnar_util.py

```diff
@@ -2623,58 +2623,66 @@ def read_coffea_root(
-    #     """
-    #     Given a parquet file located at *source*, returns a 2-tuple *(source, entries)*. Passing
-    #     *open_options* has no effect.
-    #
-    #     TODO: use read_columns?
-    #     """
-    #     return (source, pq.ParquetFile(source).metadata.num_rows)
-
-    # @classmethod
-    # def close_coffea_parquet(
-    #     cls,
-    #     source_object: str,
-    # ) -> None:
-    #     """
-    #     This is a placeholder method and has no effect.
-    #     """
-    #     return
-
-    # @classmethod
-    # def read_coffea_parquet(
-    #     cls,
-    #     source_object: str,
-    #     chunk_pos: ChunkPosition,
-    #     read_options: dict | None = None,
-    #     read_columns: set[str | Route] | None = None,
-    # ) -> coffea.nanoevents.methods.base.NanoEventsArray:
-    #     """
-    #     Given a the location of a parquet file *source_object*, returns an awkward array chunk
-    #     referred to by *chunk_pos*, assuming nanoAOD structure. *read_options* are passed to
-    #     *coffea.nanoevents.NanoEventsFactory.from_parquet*.
-    #
-    #     TODO: use read_columns?
-    #     """
-    #     # TODO: default read options? go via dak and preloaded?
-    #
-    #     # read the events chunk into memory
-    #     chunk = coffea.nanoevents.NanoEventsFactory.from_parquet(
-    #         source_object,
-    #         entry_start=chunk_pos.entry_start,
-    #         entry_stop=chunk_pos.entry_stop,
-    #         schemaclass=coffea.nanoevents.NanoAODSchema,
-    #         **(read_options or {}),
-    #     ).events()
-
-    #     return chunk
+    @classmethod
+    def open_coffea_parquet(
+        cls,
+        source: str,
+        open_options: dict | None = None,
+        read_columns: set[str | Route] | None = None,
+    ) -> tuple[str, int]:
+        """
+        Given a parquet file located at *source*, returns a 2-tuple *(source, entries)*. Passing
+        *open_options* or *read_columns* has no effect.
+        """
+        return (source, pq.ParquetFile(source).metadata.num_rows)
+
+    @classmethod
+    def close_coffea_parquet(
+        cls,
+        source_object: str,
+    ) -> None:
+        """
+        This is a placeholder method and has no effect.
+        """
+        return
+
+    @classmethod
+    def read_coffea_parquet(
+        cls,
+        source_object: str,
+        chunk_pos: ChunkPosition,
+        read_options: dict | None = None,
+        read_columns: set[str | Route] | None = None,
+    ) -> coffea.nanoevents.methods.base.NanoEventsArray:
+        """
+        Given a the location of a parquet file *source_object*, returns an awkward array chunk
+        referred to by *chunk_pos*, assuming nanoAOD structure. *read_options* are passed to
+        ``coffea.nanoevents.NanoEventsFactory.from_parquet``. *read_columns* are converted to
+        strings and, if not already present, added as nested field
+        ``parquet_options.read_dictionary`` to *read_options*.
+        """
```

Comment:
(The following comment is triggered by me currently working on other doc strings)

Can you format the doc string such that `sphinx` can automatically compile a nice overview over the parameters and the information about what is returned? So something like

```suggestion
        """
        Function to read parquet files created by :external+coffea:doc:`index` in the nanoAOD format.
        
        Given a the location of a parquet file *source_object*, returns an awkward array chunk
        referred to by *chunk_pos*, assuming nanoAOD structure. *read_options* are passed to
        :external+coffea:py:meth:`coffea.nanoevents.NanoEventsFactory.from_parquet`. *read_columns* are converted to
        strings and, if not already present, added as nested field
        ``parquet_options.read_dictionary`` to *read_options*.
        
        :param cls: The current :py:class:`~.SourceHandler` class
        :type cls: SourceHandler
        :param source_object: path to parquet file containing events to be read
        :type source_object: str
        :param chunk_pos: Position of the current chunk in the ensemble of events
        :type chunk_pos: ChunkPosition
        :param read_options: Options to consider when reading the data from *source_object*, parsed to :external+coffea:py:meth:`coffea.nanoevents.NanoEventsFactory.from_parquet`, defaults to None
        :type read_options: dict | None, optional
        :param read_columns: :py:class:`set` of columns to read from *source_object*. Converted to conform with the format expected by :external+coffea:py:meth:`coffea.nanoevents.NanoEventsFactory.from_parquet`, defaults to None
        :type read_columns: set[str | Route] | None, optional
        :return: coffea :external+coffea:py:class:`~coffea.nanoevents.methods.base.NanoEventsArray` containing events with additional behaviors attached
        :rtype: coffea.nanoevents.methods.base.NanoEventsArray
        """
```

Reply:
Similar is of course true for the other doc strings as well

Reply:
Mh, would you be ok with moving this to a dedicated PR where we adjust and prettify the doc strings? I like the newer style though (maybe except for the redundant argument type definitions which could be taken from the type hints).

Reply:
Jup, I've been working on prettifying the doc strings :D It would be good to talk about this in the next meeting with our collaborators, but I guess for this PR we can also just move ahead. Just thought we could already get started and do a uniform style in the upcoming PRs, but ok :+1: